---
- name: Converge Negative tests for 'eos_designs_facts' - missing-mlag-peer
  hosts: missing-mlag-peer
  gather_facts: false
  connection: local
  tasks:
    - name: Run failure scenario Test
      block:
        - name: Trigger Error
          ansible.builtin.import_role:
            name: arista.avd.eos_designs
      rescue:
        - name: Assert eos_designs failed with the expected error message
          ansible.builtin.assert:
            that:
              - ansible_failed_result is defined
              - ansible_failed_result.msg == expected_error_message

- name: Converge Negative tests for 'eos_designs_facts' - mlag_odd_id_oddodd1
  hosts: mlag_odd_id_oddodd1
  gather_facts: false
  connection: local
  tasks:
    - name: Run failure scenario Test
      block:
        - name: Trigger Error
          ansible.builtin.import_role:
            name: arista.avd.eos_designs
      rescue:
        - name: Assert eos_designs failed with the expected error message
          ansible.builtin.assert:
            that:
              - ansible_failed_result is defined
              - ansible_failed_result.msg == expected_error_message

- name: Converge Negative tests for 'eos_designs_facts' - mlag_same_subnet_1
  hosts: mlag_same_subnet_1
  gather_facts: false
  connection: local
  tasks:
    - name: Run failure scenario Test
      block:
        - name: Trigger Error
          ansible.builtin.import_role:
            name: arista.avd.eos_designs
      rescue:
        - name: Assert eos_designs failed with the expected error message
          ansible.builtin.assert:
            that:
              - ansible_failed_result is defined
              - ansible_failed_result.msg == expected_error_message

- name: Converge Negative tests for 'eos_designs_facts' - invalid-schema
  hosts: invalid-schema
  connection: local
  tasks:
    - name: Run failure scenario Test
      block:
        - name: Trigger Error
          ansible.builtin.import_role:
            name: arista.avd.eos_designs
      rescue:
        - name: Assert eos_designs failed with the expected error message
          ansible.builtin.assert:
            that:
              - ansible_failed_result is defined
              - ansible_failed_result.msg == expected_error_message

- name: Converge Negative tests for 'eos_designs_facts'
  hosts: invalid-schema-connected-endpoints
  connection: local
  tasks:
    - name: Run failure scenario Test
      block:
        - name: Trigger Error
          ansible.builtin.import_role:
            name: arista.avd.eos_designs
      rescue:
        - name: Assert eos_designs failed with the expected error message
          ansible.builtin.assert:
            that:
              - ansible_failed_result is defined
              - ansible_failed_result.msg == expected_error_message

- name: Converge Negative tests for 'eos_designs_facts'
  hosts: removed-schema-connected-endpoints
  connection: local
  tasks:
    - name: Run failure scenario Test
      block:
        - name: Trigger Error
          ansible.builtin.import_role:
            name: arista.avd.eos_designs
      rescue:
        - name: Assert eos_designs failed with the expected error message
          ansible.builtin.assert:
            that:
              - ansible_failed_result is defined
              - ansible_failed_result.msg == expected_error_message

- name: Converge Negative tests for 'eos_designs_facts'
  hosts: id-static-and-pool-manager
  connection: local
  tasks:
    - name: Run failure scenario Test
      block:
        - name: Trigger Error
          ansible.builtin.import_role:
            name: arista.avd.eos_designs
      rescue:
        - name: Assert eos_designs failed with the expected error message
          ansible.builtin.assert:
            that:
              - ansible_failed_result is defined
              - ansible_failed_result.msg == expected_error_message

- name: Converge Negative tests for 'eos_designs_structured_config'
  hosts: EOS_DESIGNS_FAILURES
  gather_facts: false
  connection: local
  tasks:
    - name: Run failure scenario Test
      block:
        - name: Trigger Error
          ansible.builtin.import_role:
            name: arista.avd.eos_designs
      rescue:
        - name: Assert eos_designs failed with the expected error message
          ansible.builtin.assert:
            that:
              - ansible_failed_result is defined
              - ansible_failed_result.msg == expected_error_message
